Readme: Clarify the refactor task

avivrosenberg opened the pull request:
Updated the first tasks description to clarify that we want candidates to focus on design issues.

avivrosenberg committed:
Readme: Clarify the refactor task

Updated the first tasks description to clarify that we want candidates to focus on design issues.

SimonLdj commented on README.md:
```diff
@@ -103,9 +103,13 @@ in use when testing the ships functions, not during normal operations.
-   straightforward as possible. Make sure you get the exact same output from the program as you did
+1. Refactor the code in a way that will make implementing all of the above requirements (both new and original) as easy and
+   straightforward as possible. Make sure you get the same output from the program as you did
    prior to the refactor.
```
Are you sure we can require same output? Implementing CrowsNest will most definitely change the output.

avivrosenberg commented:
Yeah but it says you should refactor, check output and only then add the
new features.
Maybe we don't need to require this, it's mainly to help them check
themselves.
On Sun, 5 Feb 2017 at 20:12 SimonLdj <notifications@github.com> wrote:

> *@SimonLdj* commented on this pull request.
> ------------------------------
>
> In README.md
> <https://github.com/SimonLdj/black-pearl-ship/pull/3#pullrequestreview-20174845>
> :
>
> > @@ -103,9 +103,13 @@ in use when testing the ships functions, not during normal operations.
>
>  Tasks for challenge completion:
>
> -1. Refactor the code in a way that will make implementing the above requirements as easy and
> -   straightforward as possible. Make sure you get the exact same output from the program as you did
> +1. Refactor the code in a way that will make implementing all of the above requirements (both new and original) as easy and
> +   straightforward as possible. Make sure you get the same output from the program as you did
>     prior to the refactor.
>
> Are you sure we can require same output? Implementing CrowsNest will most
> definitely change the output.
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/SimonLdj/black-pearl-ship/pull/3#pullrequestreview-20174845>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/AAEnd8ldGw7uvxrdC-BY5X6-hcrGWpWHks5rZhDvgaJpZM4LwoQA>
> .
>

SimonLdj commented:
I think that's not clear enough, people have asked me about that.
I think we should make it more clear (as only requirement for step 1), or remove it.

avivrosenberg commented:
Ok, I'll fix it soon.
On Sun, 5 Feb 2017 at 20:24 SimonLdj <notifications@github.com> wrote:

> Reopened #3 <https://github.com/SimonLdj/black-pearl-ship/pull/3>.
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/SimonLdj/black-pearl-ship/pull/3#event-949509490>, or mute
> the thread
> <https://github.com/notifications/unsubscribe-auth/AAEnd2JXTY_OEimzYg_rYqlBsbUHGjEcks5rZhPdgaJpZM4LwoQA>
> .
>

avivrosenberg committed:
README additional clarifications

Added explanations for the refactor task to make sure the candidates do
it first and only then continue to the next tasks.

avivrosenberg committed:
README cosmetics

avivrosenberg commented:
OK, check it out now. Let me know if there are other issues.
